src/sql/workbench/contrib/notebook/browser/cellViews/cellToolbar.component.ts

```diff
@@ -14,13 +22,78 @@ export const CELL_TOOLBAR_SELECTOR: string = 'cell-toolbar-component';
 	templateUrl: decodeURI(require.toUrl('./cellToolbar.component.html'))
 })
 export class CellToolbarComponent {
+	@ViewChild('celltoolbar', { read: ElementRef }) private celltoolbar: ElementRef;
+
 	public buttonEdit = localize('buttonEdit', "Edit");
 	public buttonClose = localize('buttonClose', "Close");
 	public buttonAdd = localize('buttonAdd', "Add new cell");
-	public buttonMoveDown = localize('buttonMoveDown', "Move cell down");
-	public buttonMoveUp = localize('buttonMoveUp', "Move cell up");
 	public buttonDelete = localize('buttonDelete', "Delete cell");
+	public buttonMoreActions = localize('buttonMoreActions', "More actions");
+
+	@Input() public cellModel: ICellModel;
+	@Input() set model(value: NotebookModel) {
+		this._model = value;
+	}
+	private _actionBar: Taskbar;
+	private _model: NotebookModel;
+	private _editCellAction: EditCellAction;
+
+	constructor(
+		@Inject(IInstantiationService) private _instantiationService: IInstantiationService,
+		@Inject(IContextMenuService) private contextMenuService: IContextMenuService
+	) {
+	}
+
+	get model(): NotebookModel {
+		return this._model;
+	}
+
+	ngOnInit() {
+		this.initActionBar();
+	}
+
+	private initActionBar(): void {
+		let addCodeCellButton = new AddCellAction('notebook.AddCodeCell', localize('codePreview', "Code cell"), 'notebook-button masked-pseudo code');
+		addCodeCellButton.cellType = CellTypes.Code;
+
+		let addTextCellButton = new AddCellAction('notebook.AddTextCell', localize('textPreview', "Markdown cell"), 'notebook-button masked-pseudo markdown');
+		addTextCellButton.cellType = CellTypes.Markdown;
+
+		let deleteButton = this._instantiationService.createInstance(CellToolbarAction, 'notebook.deleteCell', '', 'codicon masked-icon delete', this.buttonDelete, this.cellModel);
+
+		// Todo: Get this to show the list of actions specific to code or markdown cell.
+		let moreActionsButton = new MoreActions('notebook.moreActions', this.buttonMoreActions, 'codicon masked-icon more');
```

Comment:
I need this button to leverage cellToggleMoreActions.ts as dropdown content. Because the new cell toolbar will have the delete and insert cell actions represented as icons, I will need to remove the delete and insert cell actions from the actions that are pushed in CellToggleMoreActions once this is working.